LandSandBoat/server
# [LUA/SQL] Add Carpenters' Landing barge

**_I affirm:_**

## What does this pull request do?

Adds Carpenters' Landing barge interactions and transport records. Based on retail captures:

https://www.youtube.com/watch?v=1Mxu0rIvjC8
https://www.youtube.com/watch?v=JER9lZEIQrA

https://www.bg-wiki.com/ffxi/Phanauet_Channel

Each trip removes 1 use/ticket regardless of landing, if you have no ticket you'll get booted and a unique cutscene will play depending on which landing you're at. The docking order is as follows: Central > South > North > Central > South. Upon arriving at the North or South Landings the boat will drop passengers off and then exit without letting anybody on. A new boat will then appear and allow players to get on coming from the same direction that the previous boat left.

Resolves https://github.com/LandSandBoat/server/issues/97

## Steps to test these changes

North Landing    !pos -302 -2 505 2
Central Landing  !pos -137 -2 60 2
South Landing    !pos 230 -2 -530 2

1. Use one of the landing positions
2. Wait for the boat
3. Get on boat, get kicked off for not having a ticket
4. Buy a ticket
5. Get on boat again, cutscene should play and you should zone into Phanauet Channel
6. Ride the boat, cutscene will play once you zone out and you should be placed at the proper landing

## TODO

- [ ] TODO: transport.sql offset and timings are all wrong
- [ ] TODO: Verify arrive/depart animation timings once actual schedule timings solved (animations should already be correct)
- [ ] TODO: Purchasing tickets NPC interaction
- [ ] TODO: Timekeeper NPCs
- [ ] TODO: Misc NPC dialogs
- [ ] TODO: Fishing?

Looking for any assistance on figuring out how to make sense of the transport.sql timings and how the barge schedule fits into that. Currently the boats do run and you can get into Phanauet Channel but the timing is very off.
